diff --git a/model_orig/1024Mb_ddr3_parameters.vh b/model/1024Mb_ddr3_parameters.vh
index f44bab6..1ce07ff 100755
--- a/model_orig/1024Mb_ddr3_parameters.vh
+++ b/model/1024Mb_ddr3_parameters.vh
@@ -594,8 +594,8 @@
     // Simulation parameters
     parameter RZQ              =     240; // termination resistance
     parameter PRE_DEF_PAT      =   8'hAA; // value returned during mpr pre-defined pattern readout
-    parameter STOP_ON_ERROR    =       1; // If set to 1, the model will halt on command sequence/major errors
-    parameter DEBUG            =       1; // Turn on Debug messages
+    parameter STOP_ON_ERROR    =       0; // If set to 1, the model will halt on command sequence/major errors
+    parameter DEBUG            =       0; // Turn on Debug messages
     parameter BUS_DELAY        =       0; // delay in nanoseconds
     parameter RANDOM_OUT_DELAY =       0; // If set to 1, the model will put a random amount of delay on DQ/DQS during reads
     parameter RANDOM_SEED      =   31913; //seed value for random generator.
diff --git a/model_orig/2048Mb_ddr3_parameters.vh b/model/2048Mb_ddr3_parameters.vh
index 650c84a..e3f1ae1 100755
--- a/model_orig/2048Mb_ddr3_parameters.vh
+++ b/model/2048Mb_ddr3_parameters.vh
@@ -582,8 +582,8 @@
     // Simulation parameters
     parameter RZQ              =     240; // termination resistance
     parameter PRE_DEF_PAT      =   8'hAA; // value returned during mpr pre-defined pattern readout
-    parameter STOP_ON_ERROR    =       1; // If set to 1, the model will halt on command sequence/major errors
-    parameter DEBUG            =       1; // Turn on Debug messages
+    parameter STOP_ON_ERROR    =       0; // If set to 1, the model will halt on command sequence/major errors
+    parameter DEBUG            =       0; // Turn on Debug messages
     parameter BUS_DELAY        =       0; // delay in nanoseconds
     parameter RANDOM_OUT_DELAY =       0; // If set to 1, the model will put a random amount of delay on DQ/DQS during reads
     parameter RANDOM_SEED      =   31913; //seed value for random generator.
diff --git a/model_orig/4096Mb_ddr3_parameters.vh b/model/4096Mb_ddr3_parameters.vh
index 30caa69..8753cab 100755
--- a/model_orig/4096Mb_ddr3_parameters.vh
+++ b/model/4096Mb_ddr3_parameters.vh
@@ -582,8 +582,8 @@
     // Simulation parameters
     parameter RZQ              =     240; // termination resistance
     parameter PRE_DEF_PAT      =   8'hAA; // value returned during mpr pre-defined pattern readout
-    parameter STOP_ON_ERROR    =       1; // If set to 1, the model will halt on command sequence/major errors
-    parameter DEBUG            =       1; // Turn on Debug messages
+    parameter STOP_ON_ERROR    =       0; // If set to 1, the model will halt on command sequence/major errors
+    parameter DEBUG            =       0; // Turn on Debug messages
     parameter BUS_DELAY        =       0; // delay in nanoseconds
     parameter RANDOM_OUT_DELAY =       0; // If set to 1, the model will put a random amount of delay on DQ/DQS during reads
     parameter RANDOM_SEED      =   31913; //seed value for random generator.
diff --git a/model_orig/8192Mb_ddr3_parameters.vh b/model/8192Mb_ddr3_parameters.vh
index 0217758..4f653d7 100644
--- a/model_orig/8192Mb_ddr3_parameters.vh
+++ b/model/8192Mb_ddr3_parameters.vh
@@ -423,8 +423,8 @@
     // Simulation parameters
     parameter RZQ              =     240; // termination resistance
     parameter PRE_DEF_PAT      =   8'hAA; // value returned during mpr pre-defined pattern readout
-    parameter STOP_ON_ERROR    =       1; // If set to 1, the model will halt on command sequence/major errors
-    parameter DEBUG            =       1; // Turn on Debug messages
+    parameter STOP_ON_ERROR    =       0; // If set to 1, the model will halt on command sequence/major errors
+    parameter DEBUG            =       0; // Turn on Debug messages
     parameter BUS_DELAY        =       0; // delay in nanoseconds
     parameter RANDOM_OUT_DELAY =       0; // If set to 1, the model will put a random amount of delay on DQ/DQS during reads
     parameter RANDOM_SEED      =   31913; //seed value for random generator.
